src/averageAges.js

```diff
@@ -55,7 +93,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let count = 0;
+
+  const childAndMoms = people.reduce((perv, child) => {
+    return perv + people.reduce((prev, mom) => {
+      const checking = child.mother === mom.name
+        && ((onlyWithSon && child.sex === 'm') || !onlyWithSon);
+
+      if (checking) {
```

Comment:
Replace the `if` statement with a ternary operator for better readability. Example: `return checking ? prev + difference : prev;`